app-backend/database/src/main/scala/com/azavea/rf/database/tables/Buckets.scala

```diff
@@ -239,4 +282,40 @@ object Buckets extends TableQuery(tag => new Buckets(tag)) with LazyLogging {
       sceneBucketJoinQuery.delete
     }
   }
+
+  /** Removes scenes from bucket
+    *
+    * @param sceneIds Seq[UUID] primary key of scenes to remove from bucket
+    * @param bucketId UUID primary key of bucket these scenes will be removed from
+    */
+  def deleteScenesFromBucket(sceneIds: Seq[UUID], bucketId: UUID)
+    (implicit database: DB): Future[Int] = {
+
+    val sceneBucketJoinQuery = for {
+      s <- ScenesToBuckets if s.sceneId.inSet(sceneIds) && s.bucketId === bucketId
+    } yield s
+
+    database.db.run {
+      sceneBucketJoinQuery.delete
+    }
+  }
+
+  /** Removes all scenes from bucket, then adds all specified scenes to it
+    *
+    * @param sceneIds Seq[UUID] primary key of scenes to add to bucket
+    * @param bucketId UUID primary key of bucket to remove all scenes from and add specified scenes to
+    * @return Scenes that were added
+    */
+  def replaceScenesInBucket(sceneIds: Seq[UUID], bucketId: UUID)
+    (implicit database: DB): Future[Iterable[Scene.WithRelated]] = {
+
+    database.db.run {
+      (for {
+        _ <- ScenesToBuckets.filter(sb => sb.bucketId === bucketId).delete
+        _ <- ScenesToBuckets ++= sceneIds.map(sceneId => SceneToBucket(sceneId, bucketId))
+      } yield()).transactionally
```

Comment:
Using for comprehension here actually entails interaction you don't intend. Using `map` and `flatMap` on `DBIO` action requires getting an implicit `ExecutionContext` because you could be using a result of one query to run the next. Since you're mapping to `_` you obviously don't need that.

I think you should be able to follow this pattern here:

```scala
    val scene = sceneCreate.toScene(user.id)
    val thumbnails = sceneCreate.thumbnails.map(_.toThumbnail(user.id, scene))
    val images = sceneCreate.images.map(_.toImage(user.id, scene))

    val actions = Scenes.forceInsert(scene)
      .andThen(DBIO.seq(thumbnails.map(Thumbnails.forceInsert): _*))
      .andThen(DBIO.seq(images.map(Images.forceInsert): _*))

    database.db.run {
      actions.transactionally
    } map { x =>
      scene.withRelatedFromComponents(images, thumbnails)
    }
```